presto-main/src/main/java/com/facebook/presto/sql/planner/SortExpressionExtractor.java

```diff
@@ -59,7 +63,34 @@ private SortExpressionExtractor() {}
             return Optional.empty();
         }
 
-        return new SortExpressionVisitor(buildSymbols).process(filter);
+        List<Expression> filterConjuncts = ExpressionUtils.extractConjuncts(filter);
+        SortExpressionVisitor visitor = new SortExpressionVisitor(buildSymbols);
+
+        List<SortExpressionContext> sortExpressionCandidates = filterConjuncts.stream()
+                .map(visitor::process)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(groupingBy(SortExpressionContext::getSortExpression, reducing(SortExpressionExtractor::merge)))
+                .values()
+                .stream()
+                .map(Optional::get)
```

Comment:
This looks suspicious: call to `Optional.get` without `.ifPresent`. You can avoid having those misleading `Optional`-s, if you replace `.collect(groupingBy(...))` with `.collect(toMap(SortExpressionContext::getSortExpression, c->c, SortExpressionExtractor::merge)`.